plugins/buildingplan/buildingplan_cycle.cpp

```diff
@@ -45,13 +46,38 @@ struct BadFlags {
     }
 };
 
+// cache of walkability groups that have citizens in them
+static std::unordered_set<uint16_t> accessible_walkability_groups;
+
+void update_walkability_groups(){
+    static int32_t update_timestamp = -1;
+
+    // ensure that we update at most once per tick
+    auto frame_counter = df::global::world->frame_counter;
+    if (frame_counter == update_timestamp)
+        return;
+    else
+        update_timestamp = frame_counter;
+
+    accessible_walkability_groups.clear();
+    auto get_group = [](df::unit *unit){
```

Comment:
I may be wrong, but doesn't this need to be `[&]` to refer to `accessible_walkability_groups` by reference instead of a copy?

Reply:
GCC started yelling at me when I was trying to capture a variable with static storage duration. This is not being captured by copy, it is not being captured at all. So, I think you are wrong ;)

Reply:
ok, that makes sense.